bittorrent/params_test.go

```diff
@@ -97,33 +97,35 @@ func TestParseShouldNotPanicURLData(t *testing.T) {
 }
 
 func BenchmarkParseQuery(b *testing.B) {
-	announceStrings := make([]string, 0)
+	announceStrings := make([]string, len(ValidAnnounceArguments))
 	for i := range ValidAnnounceArguments {
 		announceStrings = append(announceStrings, ValidAnnounceArguments[i].Encode())
 	}
-	b.ResetTimer()
-	for bCount := 0; bCount < b.N; bCount++ {
-		i := bCount % len(announceStrings)
-		parsedQueryObj, err := parseQuery(announceStrings[i])
-		if err != nil {
-			b.Error(err, i)
-			b.Log(parsedQueryObj)
+	b.Run("ParseQuery", func(b *testing.B) {
+		for bCount := 0; bCount < b.N; bCount++ {
+			i := bCount % len(announceStrings)
+			parsedQueryObj, err := parseQuery(announceStrings[i])
+			if err != nil {
+				b.Error(err, i)
+				b.Log(parsedQueryObj)
+			}
 		}
-	}
+	})
 }
 
 func BenchmarkURLParseQuery(b *testing.B) {
-	announceStrings := make([]string, 0)
+	announceStrings := make([]string, len(ValidAnnounceArguments))
```

Comment:
this is setting the len and not the cap

Reply:
Oh right, so we get 50% empty strings, which would probably be very fast to parse... Same above